### PR TITLE
feat(Dropdown): add option for divider in dropdown

### PR DESCRIPTION
**Pull request by rivka-ungar**
https://monday.monday.com/boards/3532714909/views/80492480/pulses/7178692867

Adding option for divider in dropdown

Screenshot 2024-09-08 at 15 40 44

### CODE REVIEW

File: packages/core/src/components/Dropdown/Dropdown.styles.js [hunk 1]
```diff
@@ -358,15 +358,32 @@ const indicatorSeparator = () => () => ({
...
+    ...(withGroupDivider && {
+      ":not(:last-child)": {
+        position: "relative",
+        paddingBottom: "8px",
+        "::after": {
+          content: '""',
+          position: "absolute",
+          bottom: "0",
+          left: "50%",
+          transform: "translateX(-50%)",
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/Dropdown.styles.js:375]:
```suggestion
          left: var(--spacing-small),
```

Can also work

File: packages/core/src/components/Dropdown/Dropdown.styles.js [hunk 2]
```diff
@@ -358,15 +358,32 @@ const indicatorSeparator = () => () => ({
...
+          width: `calc(100% - (${getCSSVar("spacing-small")} * 2))`,
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/Dropdown.styles.js:376]:
```suggestion
          width: `calc(100% - (var(--spacing-small) * 2))`,
```
I'm not 100% sure about css-in-js, but shouldn't this work?

File: packages/core/src/components/Dropdown/Dropdown.styles.js [hunk 3]
```diff
@@ -358,15 +358,32 @@ const indicatorSeparator = () => () => ({
...
+          height: "1px",
+          backgroundColor: getCSSVar("layout-border-color")
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/Dropdown.styles.js:378]:
```suggestion
          backgroundColor: "var(layout-border-color)"
```
same as the above comments, shouldn't this work?

File: packages/core/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -110,7 +110,8 @@ const Dropdown: VibeComponent<DropdownComponentProps, HTMLElement> & {
       popupsContainerSelector,
       filterOption,
       menuPosition = Dropdown.menuPositions.ABSOLUTE,
-      "data-testid": dataTestId
+      "data-testid": dataTestId,
+      withGroupDivider = false
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/Dropdown.tsx:114]: do we have any way to put it in the supplied options array instead of as a prop?

**Review by YossiSaadi — COMMENTED** [review]
nice solution, had a few thoughts and questions, lmk what you think

File: packages/core/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (rivka-ungar) [reply packages/core/src/components/Dropdown/Dropdown.tsx:114]: We wanted to keep it similar to the combobox api that has `withCategoriesDivider`. Also if it is in the options array will have to send the prop in each group separately.

Comment (YossiSaadi) [reply packages/core/src/components/Dropdown/Dropdown.tsx:114]: oh you're right
good call 💪🏼

**Review by YossiSaadi — APPROVED** [review]
💯

**Pull request merged**